src/swift/Source.swift

```diff
@@ -144,132 +150,151 @@ public extension DispatchSource {
-		return __dispatch_source_create(
-			_swift_dispatch_source_type_memorypressure(), 0, eventMask.rawValue, queue) as DispatchSourceMemoryPressure
+		let source = dispatch_source_create(_swift_dispatch_source_type_memorypressure(), 0, eventMask.rawValue, queue.__wrapped)
+		return DispatchSourceMemoryPressure(source)
 	}
+#endif
 
+#if HAVE_MACH
 	public class func process(identifier: pid_t, eventMask: ProcessEvent, queue: DispatchQueue? = nil) -> DispatchSourceProcess {
-		return __dispatch_source_create(
-			_swift_dispatch_source_type_proc(), UInt(identifier), eventMask.rawValue, queue) as DispatchSourceProcess
+		let source = dispatch_source_create(_swift_dispatch_source_type_proc(), UInt(identifier), eventMask.rawValue, queue?.__wrapped)
+		return DispatchSource(source: source) as DispatchSourceProcess
 	}
+#endif
 
 	public class func read(fileDescriptor: Int32, queue: DispatchQueue? = nil) -> DispatchSourceRead {
-		return __dispatch_source_create(
-			_swift_dispatch_source_type_read(), UInt(fileDescriptor), 0, queue) as DispatchSourceRead
+		let source = dispatch_source_create(_swift_dispatch_source_type_read(), UInt(fileDescriptor), 0, queue?.__wrapped)
+		return DispatchSource(source: source) as DispatchSourceRead
 	}
 
 	public class func signal(signal: Int32, queue: DispatchQueue? = nil) -> DispatchSourceSignal {
-		return __dispatch_source_create(
-			_swift_dispatch_source_type_read(), UInt(signal), 0, queue) as DispatchSourceSignal
+		let source = dispatch_source_create(_swift_dispatch_source_type_signal(), UInt(signal), 0, queue?.__wrapped)
+		return DispatchSource(source: source) as DispatchSourceSignal
 	}
 
 	public class func timer(flags: TimerFlags = [], queue: DispatchQueue? = nil) -> DispatchSourceTimer {
-		return __dispatch_source_create(_swift_dispatch_source_type_timer(), 0, flags.rawValue, queue) as DispatchSourceTimer
+		let source = dispatch_source_create(_swift_dispatch_source_type_timer(), 0, flags.rawValue, queue?.__wrapped)
+		return DispatchSource(source: source) as DispatchSourceTimer
 	}
 
 	public class func userDataAdd(queue: DispatchQueue? = nil) -> DispatchSourceUserDataAdd {
-		return __dispatch_source_create(_swift_dispatch_source_type_data_add(), 0, 0, queue) as DispatchSourceUserDataAdd
+		let source = dispatch_source_create(_swift_dispatch_source_type_data_add(), 0, 0, queue?.__wrapped)
+		return DispatchSource(source: source) as DispatchSourceUserDataAdd
 	}
 
 	public class func userDataOr(queue: DispatchQueue? = nil) -> DispatchSourceUserDataOr {
-		return __dispatch_source_create(_swift_dispatch_source_type_data_or(), 0, 0, queue) as DispatchSourceUserDataOr
+		let source = dispatch_source_create(_swift_dispatch_source_type_data_or(), 0, 0, queue?.__wrapped)
+		return DispatchSource(source: source) as DispatchSourceUserDataOr
 	}
 
 	public class func fileSystemObject(fileDescriptor: Int32, eventMask: FileSystemEvent, queue: DispatchQueue? = nil) -> DispatchSourceFileSystemObject {
```

Comment:
I would remove this from linux as my understanding is that the semantics of inotify & vnode sources isn't compatible, or I would make sure that it can work